File: BaiduMapTest/PlaceSearchTest.cs
using System;
using System.Collections.Generic;
using BaiduMap.Response.Place;
using BaiduMap.Util;
using Xunit;
using System.Linq;
using System.Web;
using Shouldly;
using BaiduMap.Request;
using BaiduMap.Request.Models;

namespace BaiduMapTest
{
    public class PlaceSearchTest
    {
        [Fact]
        public void ExecuteRegionSearch()
        {
            var baiduClient = ClientCreator.Create();
            var regionModel = new PlaceRegionModel
            {
                Query = "ATM机",
                Tag = "银行",
                Region = "杭州"
            };
            var req = new PlaceSearchRequest<PlaceRegionModel>(regionModel);
            PlaceSearchResponse resp = baiduClient.Execute(req);
            resp.Status.ShouldBe(0);
            resp.Message.ShouldBe("ok");
            resp.Results.Count.ShouldBe(10);
        }

        [Fact]
        public void ExecuteCircumSearch()
        {
            var baiduClient = ClientCreator.Create();
            var circumModel = new PlaceCircumModel
            {
                Query = "银行",
                Location = "39.915,116.404",
                Radius = "2000"
            };
            var req = new PlaceSearchRequest<PlaceCircumModel>(circumModel);
            PlaceSearchResponse resp = baiduClient.Execute(req);
            resp.Status.ShouldBe(0);
            resp.Message.ShouldBe("ok");
        }

        [Fact]
        public void ExecuteRectangeSearch()
        {
            var baiduClient = ClientCreator.Create();
            var rectangeModel = new PlaceRectangeModel
            {
                Query = "银行",
                Bounds = "39.915,116.404,39.975,116.414"
            };
            var req = new PlaceSearchRequest<PlaceRectangeModel>(rectangeModel);
            PlaceSearchResponse resp = baiduClient.Execute(req);
            resp.Status.ShouldBe(0);
            resp.Message.ShouldBe("ok");
        }
    }
}

File: JAVA/baidumap-webapi/src/test/java/wang/seamas/baidumap/PlaceSearchTest.java
package wang.seamas.baidumap;

import org.junit.Assert;
import org.junit.Test;
import wang.seamas.baidumap.request.PlaceSearchRequest;
import wang.seamas.baidumap.request.model.PlaceCircumModel;
import wang.seamas.baidumap.request.model.PlaceRectangeModel;
import wang.seamas.baidumap.request.model.PlaceRegionModel;
import wang.seamas.baidumap.response.PlaceSearchResponse;

/**
 * @author: Seamas Wang
 * @description:
 * @date: Created in 11:59 2018/8/6
 */
public class PlaceSearchTest {

    @Test
    public void ExecuteRegionSearch()
    {
        BaiduApiClient baiduClient = ClientCreator.Create();
        PlaceRegionModel model = new PlaceRegionModel();
        model.setQuery("ATM机");
        model.setTag("银行");
        model.setRegion("杭州");

        PlaceSearchRequest<PlaceRegionModel> req = new PlaceSearchRequest<PlaceRegionModel>(model);
        PlaceSearchResponse resp = baiduClient.execute(req);
        Assert.assertEquals(resp.getStatus(), 0);
    }

    @Test
    public void ExecuteCircumSearch()
    {
        BaiduApiClient baiduClient = ClientCreator.Create();
        PlaceCircumModel model = new PlaceCircumModel();
        model.setQuery("银行");
        model.setLocation("39.915,116.404");
        model.setRadius("2000");

        PlaceSearchRequest<PlaceCircumModel> req = new PlaceSearchRequest<>(model);
        PlaceSearchResponse resp = baiduClient.execute(req);
        Assert.assertEquals(resp.getStatus(), 0);
    }

    @Test
    public void ExecuteRectangeSearch()
    {
        BaiduApiClient baiduClient = ClientCreator.Create();
        PlaceRectangeModel model = new PlaceRectangeModel();
        model.setQuery("银行");
        model.setBounds("39.915,116.404,39.975,116.414");

        PlaceSearchRequest<PlaceRectangeModel> req = new PlaceSearchRequest<>(model);
        PlaceSearchResponse resp = baiduClient.execute(req);
        Assert.assertEquals(resp.getStatus(), 0);
    }
}
